cross-chain-atomic-swap-cordapp/src/main/kotlin/jp/co/layerx/cordage/crosschainatomicswap/flow/EventWatchFlow.kt

```diff
@@ -0,0 +1,136 @@
+        val web3: Web3j = Web3j.build(HttpService(ETHEREUM_RPC_URL))
+
+        val eventMapping = mapOf<String, Event>("Locked" to Settlement.LOCKED_EVENT)
+
+        val swapDetailType = Arrays.asList<TypeReference<*>?>(
+            object : TypeReference<Address?>() {},
+            object : TypeReference<Address?>() {},
+            object : TypeReference<Uint256?>() {},
+            object : TypeReference<Uint256?>() {},
+            object : TypeReference<Uint8?>() {}
+        )
+
+        object CREATING_WATCHERSTATE: ProgressTracker.Step("Creating new WatcherState.")
+        object WATCHING_EVENT: ProgressTracker.Step("Getting Ethereum Events.")
+        object GENERATING_TRANSACTION : ProgressTracker.Step("Generating a WatcherState transaction.")
+        object VERIFYING_TRANSACTION : ProgressTracker.Step("Verifying a WatcherState transaction.")
+        object SIGNING_TRANSACTION : ProgressTracker.Step("Signing transaction with our private key.")
+        object FINALISING_TRANSACTION : ProgressTracker.Step("Recording transaction.") {
+            override fun childProgressTracker() = FinalityFlow.tracker()
+        }
+
+        fun tracker() = ProgressTracker(
+                CREATING_WATCHERSTATE,
+                WATCHING_EVENT,
+                GENERATING_TRANSACTION,
+                VERIFYING_TRANSACTION,
+                SIGNING_TRANSACTION,
+                FINALISING_TRANSACTION
+        )
+    }
+
+    override val progressTracker = tracker()
+
+    @Suspendable
+    override fun call(): String {
+        progressTracker.currentStep = WATCHING_EVENT
+        val input = serviceHub.toStateAndRef<WatcherState>(stateRef)
+        val fromBlockNumber = input.state.data.fromBlockNumber
+        val toBlockNumber = input.state.data.toBlockNumber
+        val targetContractAddress = input.state.data.targetContractAddress
+        val eventName = input.state.data.eventName
+        val proposalStateAndRef = input.state.data.proposalStateAndRef
+        val proposalState = proposalStateAndRef.state.data
+        val searchId = proposalState.swapId
+        val event = eventMapping[eventName]
+
+        val filter = EthFilter(DefaultBlockParameter.valueOf(fromBlockNumber),
+                DefaultBlockParameter.valueOf(toBlockNumber),
+                targetContractAddress)
+
+        val ethLogs = web3.ethGetLogs(filter).send()
+
+        val decodedLogs = ethLogs.result?.map { (it.get() as Log).data }
+                ?.map { DefaultFunctionReturnDecoder.decode(it, event?.nonIndexedParameters) }
+        if (decodedLogs != null && decodedLogs.isNotEmpty()) {
+            decodedLogs.forEach { abiTypes ->
+                // find event values by searchId
+                val eventValues = abiTypes?.map { it.value }
+                if (eventValues != null && eventValues.isNotEmpty()) {
+                    val lockedEvent = eventValues?.let { LockedEvent.listToLockedEvent(it) }
```

Comment:
Since eventValues was nullable, I used let to call listToLockedEvent, but since I put a null check, as you said, I don't need to use let here.
Thank you so much! Corrected! 
b2a5a33